tests/nested/manual/hybrid-remodel/task.yaml

```diff
@@ -0,0 +1,61 @@
+summary: Test remodeling on a hybrid system
+
+details: |
+  This test remodels on a hybrid system to install a new kernel snap and new
+  application snaps.
+
+systems: [ubuntu-22.04-64]
+
+prepare: |
+  # FIXME: The gadget snap for classic is not yet available. So for
+  #        the moment, the test modifies the gadget for Ubuntu Core.
+  #        This should be removed when the classic snap is available.
+  VERSION="$(tests.nested show version)"
+  snap download --basename=pc --channel="$VERSION/edge" pc
+  unsquashfs -d pc-gadget pc.snap
+
+  # gadget.yaml needs a different structure than for Ubuntu Core
+  python3 "$TESTSLIB"/tweak-gadget.py pc-gadget/meta/gadget.yaml
+  snap pack --filename=pc_x1.snap pc-gadget
+
+  SNAPD_DEB="$(find "$PROJECT_PATH"/.. -maxdepth 1 -name "snapd_*.deb" -printf "%f\n")"
+  SNAP_DEB_PARAM="--snapd-deb $PROJECT_PATH/../$SNAPD_DEB"
+
+  # create an image that looks like a classic image
+  # shellcheck disable=SC2086
+  "$TESTSTOOLS"/mkimage-uc22 --base-dir "$PWD" --snap pc_x1.snap $SNAP_DEB_PARAM "$TESTSLIB"/assertions/classic-model.assert
+
+  tests.pkgs install qemu qemu-utils genisoimage sshpass qemu-kvm cloud-image-utils ovmf kpartx cpu-checker
+
+  MACHINE_PARAM="-machine ubuntu"
+  if kvm-ok; then
+    MACHINE_PARAM="-machine ubuntu,accel=kvm"
+  fi
+
+  tests.systemd create-and-start-unit "nested-vm" "qemu-system-x86_64 -m 1500 -nographic $MACHINE_PARAM -snapshot -netdev user,id=net.0,hostfwd=tcp::10022-:22 -device rtl8139,netdev=net.0 -bios /usr/share/OVMF/OVMF_CODE.fd -drive file=$PWD/boot.img,if=virtio -serial file:$PWD/serial.log"
+
+  # run built image
+  remote.setup config --host localhost --port 10022 --user user1 --pass ubuntu
+  remote.wait-for ssh
+
+  # wait until the image boots and seeds  
+  remote.exec "sudo snap wait system seed.loaded"
+
+restore: |
+  tests.systemd stop-unit --remove "nested-vm"
+
+execute: |
+  remote.wait-for device-initialized
+
+  remote.push "${TESTSLIB}/assertions/classic-model-rev1.assert"
+
+  # remodel and reboot. we need to reboot because we swapped the kernel snap
+  remote.exec 'sudo snap remodel classic-model-rev1.assert'
+  remote.exec 'test -f /run/reboot-required'
+  remote.exec 'snap changes' | MATCH 'Wait.*Refresh model assertion from revision 0 to 1'
+  remote.exec 'sudo reboot || true'
+
+  retry -n 30 --wait 1 sh -c "remote.exec 'snap changes' | MATCH 'Done.*Refresh model assertion from revision 0 to 1'"
+
+  remote.exec 'snap list jq-core22'
+  remote.exec 'snap list pc-kernel' | awk 'NR != 1 { print $4 }' | MATCH '22-hwe/stable'
```

Comment:
this will create a recovery system?  do we need to test something about it? or should we leave a TODO and add something with the work about setting those up in the first place with hybrid systems?

Reply:
In this test, the gadget uses `system-seed-null`, so there won't be a recovery system.